### ISSUE TITLE
Visitors from `visitor-plugin-common` incompatible with `Visitor` from `graphql`

### ISSUE THREAD

**Issue by vhenzl** [dotansimha/graphql-code-generator#1952]
The second parameter of `VisitFn` can be `undefined` (see [TS](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/graphql/language/visitor.d.ts#L31) and [Flow](https://github.com/graphql/graphql-js/blob/master/src/language/visitor.js#L39) declarations), but visitors in `visitor-plugin-common` declare `key` as `number | string` only.

If you try to use some of these visitors in a project with `"strictNullChecks": true` (or `"strict": true`) configuration, it will results in errors like this one:
```
TS2345: Argument of type '{ leave: MyVisitor<{}, ParsedTypesConfig>; }' is not assignable to parameter of type 'Visitor<ASTKindToNode, ASTNode>'.
  Type '{ leave: MyVisitor<{}, ParsedTypesConfig>; }' is not assignable to type 'EnterLeave<VisitFn<ASTNode, ASTNode> | { Name?: VisitFn<ASTNode, NameNode> | undefined; Document?: VisitFn<ASTNode, DocumentNode> | undefined; OperationDefinition?: VisitFn<ASTNode, OperationDefinitionNode> | undefined; ... 39 more ...; InputObjectTypeExtension?: VisitFn<...> | undefined; }>'.
    Types of property 'leave' are incompatible.
      Type 'MyVisitor<{}, ParsedTypesConfig>' is not assignable to type 'VisitFn<ASTNode, ASTNode> | { Name?: VisitFn<ASTNode, NameNode> | undefined; Document?: VisitFn<ASTNode, DocumentNode> | undefined; OperationDefinition?: VisitFn<ASTNode, OperationDefinitionNode> | undefined; ... 39 more ...; InputObjectTypeExtension?: VisitFn<...> | undefined; } | undefined'.
        Type 'MyVisitor<{}, ParsedTypesConfig>' is not assignable to type '{ Name?: VisitFn<ASTNode, NameNode> | undefined; Document?: VisitFn<ASTNode, DocumentNode> | undefined; OperationDefinition?: VisitFn<ASTNode, OperationDefinitionNode> | undefined; ... 39 more ...; InputObjectTypeExtension?: VisitFn<...> | undefined; }'.
          Types of property 'ObjectTypeDefinition' are incompatible.
            Type '(node: ObjectTypeDefinitionNode, key: string | number, parent: any) => string' is not assignable to type 'VisitFn<ASTNode, ObjectTypeDefinitionNode>'.
              Types of parameters 'key' and 'key' are incompatible.
                Type 'string | number | undefined' is not assignable to type 'string | number'.
                  Type 'undefined' is not assignable to type 'string | number'.
```

Exported types should be compatible with `graphql` package so they can be really reused in other projects. 

Also, I think `visitor-plugin-common` and possibly other (`plugin-helpers`, ?) packages, which are meant for writing custom extensions, should have `"strict": true` option to prevent incompatibilities like this one.

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/2266

**Comment by dotansimha** [comment]
Fixed in 1.5.0 🎉
